internal/cli/login/aap.go

```diff
@@ -0,0 +1,107 @@
+package login
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/openshift/osincli"
+)
+
+type AAPOAuth struct {
+	ClientId           string
+	CAFile             string
+	InsecureSkipVerify bool
+	ConfigUrl          string
+}
+
+type AAPRoundTripper struct {
+	Transport http.RoundTripper
+}
+
+func (c *AAPRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := c.Transport.RoundTrip(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// AAPGateway returns 201 on success, but osincli expects 200
+	if resp.StatusCode == 201 {
```

Comment:
package http has constants for these values

Reply:
added